src/views/Give/hooks/useEditGive.ts

```diff
@@ -6,74 +6,126 @@ import { GIVE_ADDRESSES, SOHM_ADDRESSES } from "src/constants/addresses";
 import { useWeb3Context } from "src/hooks";
 import { balanceQueryKey } from "src/hooks/useBalance";
 import { useDynamicGiveContract } from "src/hooks/useContract";
+import { donationInfoQueryKey } from "src/hooks/useGiveInfo";
 import { useTestableNetworks } from "src/hooks/useTestableNetworks";
 import { error as createErrorToast, info as createInfoToast } from "src/slices/MessagesSlice";
 
 import { GiveData } from "../Interfaces";
+import { IUAData, trackGiveEvent } from "../utils/googleAnalytics";
 
+/**
+ * @notice Increases the value of an active donation
+ * @returns ContractReceipt for the deposit
+ */
 export const useIncreaseGive = () => {
   const dispatch = useDispatch();
   const client = useQueryClient();
   const { address } = useWeb3Context();
   const networks = useTestableNetworks();
   const contract = useDynamicGiveContract(GIVE_ADDRESSES, true);
 
+  // Mutation to interact with the YieldDirector contract
   return useMutation<ContractReceipt, Error, GiveData>(
+    // Pass in an object with an amount and a recipient parameter
     async ({ amount: amount_, recipient: recipient_ }) => {
+      // Validate inputs
       if (parseFloat(amount_) <= 0) throw new Error(t`An increase Give amount must be positive`);
 
-      if (!contract) throw new Error(t`Please switch to the Ethereum network to donate yield`);
+      // Confirm that the user is on a chain where YieldDirector exists
+      if (!contract)
+        throw new Error(
+          t`Give is not supported on this network. Please switch to a supported network, such as Ethereum mainnet`,
+        );
 
+      // Create transaction to deposit passed amount to the passed recipient
       const transaction = await contract.deposit(ethers.utils.parseUnits(amount_, "gwei"), recipient_);
+
+      const uaData: IUAData = {
```

Comment:
I pushed some changes a few weeks back that logs a GA event before and after the transaction (as we were missing events and wanted to find out why). Can you ensure this is replicated here?

https://github.com/OlympusDAO/olympus-frontend/pull/1574/files